src/main/java/org/elasticsearch/common/lucene/all/AllTermQuery.java

```diff
@@ -51,7 +51,7 @@ public AllTermQuery(Term term) {
     }
 
     @Override
-    public Weight createWeight(IndexSearcher searcher, boolean needsScores) throws IOException {
+    public SpanWeight createWeight(IndexSearcher searcher, boolean needsScores) throws IOException {
```

Comment:
This class overrides weight/scorer, where is its positions check (throw IAE if proximity is not enabled) ?